main.go

```diff
@@ -62,24 +58,17 @@ func clusterProvider(cluster clusterplugin.Cluster) yip.YipConfig {
 		systemName = agentSystemName
 	}
 
-	_config, _ := config.Scan(config.Directories(configScanDir...))
-
-	if _config != nil {
-		for _, e := range _config.Env {
-			pair := strings.SplitN(e, "=", 2)
-			if len(pair) >= 2 {
-				os.Setenv(pair[0], pair[1])
-			}
-		}
-	}
-
 	var providerConfig bytes.Buffer
 	_ = yaml.NewEncoder(&providerConfig).Encode(&k3sConfig)
 
 	userOptions, _ := kyaml.YAMLToJSON([]byte(userOptionConfig))
 	proxyOptions, _ := kyaml.YAMLToJSON([]byte(cluster.Options))
 	options, _ := kyaml.YAMLToJSON(providerConfig.Bytes())
 
+	for key, value := range cluster.Env {
+		os.Setenv(key, value)
```

Comment:
Setting this here will have no effect on k3s this binary is called as a subprocess.  We should be passing these variables to the k3s configuration.

Reply:
I set them here, read them in the proxyEnv function and set the content in the k3s configuration. 
The k3s configuration contains the proxy attributes with values.